src/system/SystemLayer.cpp

```diff
@@ -297,13 +296,84 @@ Error Layer::StartTimer(uint32_t aMilliseconds, TimerCompleteFunct aComplete, vo
     SuccessOrExit(lReturn);
```

Comment:
Can we remove to 'goto exit' pattern here? A goto exit that just returns nullptr is harder to read than basic returns.

Reply:
Acked